src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+ */
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvironmentModel;
+import com.troubleshooting.model.EnvironmentsModel;
+import com.troubleshooting.model.LogsModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public final class ContainerDataRetrieval implements IDataRetrieval {
+    private ContainerDataAccess access;
+    private ConjurLogModel log;
+    private LogsModel logs;
+    private EnvironmentModel env;
+    private EnvironmentsModel envs;
+
+    static String STDERR = "STDERR";
+
+    @Autowired
+    public ContainerDataRetrieval(ContainerDataAccess access, ConjurLogModel log, LogsModel logs, EnvironmentModel env, EnvironmentsModel envs) {
+        this.access = access;
+        this.log = log;
+        this.logs = logs;
+        this.env = env;
+        this.envs = envs;
+    }
+
+    @Override
+    public LogsModel getLogs(String containerID, String query) {
+        List<String> loggingFrames = new ArrayList<String>();
+
+        try {
+            loggingFrames = access.getLogs(containerID, query);
+
+            loggingFrames.stream().forEach(entry -> extractImportantPartsOfLogEntry(entry));
+        } catch (NullPointerException e) {
+            System.out.println(e);
+        }
+        return logs;
+    }
+
+    private void extractImportantPartsOfLogEntry(String logInstance) {
+        String[] logEntry = logInstance.split(" ", 5);
```

Comment:
this calls out for a parser.
Create a class LogEntry, add a static method to it: static LogEntry fromString(String), there you will handle the parsing logic, the class will have all the fields (e.g. origin, requestId, etc)

Reply:
I have a class like that, called ConjurLogModel and it has the following
```
public class ConjurLogModel implements ILogModel {
    private String origin;
    private String request_id;
    private String thread_id;
    private String message;
```

In terms of the parser, it will have just this
```
String[] logEntry = logInstance.split(" ", 5);
```
or also the logic for filling the class fields (origin, requestId, etc)?

Reply:
also why static?

Reply:
the parser will encapsulate the parsing logic of the raw input, thus of the input changes the only change you will need to do is in the parser. 

Note: request_id that is ruby naming not java.
also why static? It is a factory method so to speak.
https://tinyurl.com/y9zf2983